### PR TITLE
Add long short example

### PR DESCRIPTION
**Pull request by andrewkim316**
Adds long-short equity trading algorithm to golang SDK

### CODE REVIEW

File: examples/mean-reversion.go [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+	runningAverage float64
+	lastOrder      string
+	timeToClose    float64
+	stock          string
+}
+
+var alpacaClient alpacaClientContainer
+
+func init() {
+	os.Setenv(common.EnvApiKeyID, "PKM4OPMZT6GJFCLF9ZCB")
```
Comment (ttt733) [inline examples/mean-reversion.go]: These are just paper keys, but you should probably at least regenerate them and remove them from public stuff like this.

File: examples/README.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Examples
+
+This directory contains example trading algorithms that connect to the paper-trading API.  These scripts are meant to be run in a Node.js app, where you first install the Alpaca node module, then run the trading algorithm.  Please note you will need to replace the `API_KEY` and `API_SECRET` parameters at the top of the file with your own information from the [Alpaca dashboard](https://app.alpaca.markets/).  Please also note that the performance of these scripts in a real trading environment is not guaranteed. While they are written with the goal of showing realistic uses of the SDK, there is no guarantee that the strategies they outline are a good fit for your own brokerage account.
```
Comment (ttt733) [inline examples/README.md]: This needs to be changed from Node to Go instructions.

File: examples/README.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+## Mean Reversion
+
+This trading algorithm bases its strategy on a mean reversion theory, which essentially in the long run, the stock price will correct to the mean.  This means we'd want to execute trades when the stock price is below the running average, as the theory states that the stock price will eventually rise to the mean.  The algorithm does this by taking the 20 minute running average stock price of a given stock (in this case "AAPL") and longs or sells based on the average.  After every minute, the algorithm will re-evaluate the mean and see if adjustments to the position need to be made.  For more information on this strategy, you can read [this link](https://medium.com/automation-generation/a-simple-mean-reversion-stock-trading-script-in-c-fdd3d147af95) detailing a mean reversion strategy in C#.
```
Comment (ttt733) [inline examples/README.md]: "which essentially guesses that the stock price will correct to the mean" would be better IMO.

File: examples/README.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+## Long-Short Equity
+
+This trading algorithm implements the long-short equity strategy.  This means that the algorithm will rank a given universe of stocks based on a certain metric, and long the top ranked stocks and short the lower ranked stocks.  More specifically, the algorithm uses the frequently used 130/30 percent equity split between longs and shorts (130% of equity used for longs, 30% of equity used for shorts).  The algorithm will then grab the top and bottom 25% of stocks, and long or short them accordingly.  The algorithm will purchase equal quantities across a bucket of stocks, so all stocks in the long bucket are ordered with the same quantity (same with the short bucket).  After every minute, the algorithm will re-rank the stocks and make adjustments to the position if necessary.
```
Comment (ttt733) [inline examples/README.md]: Could you maybe link to an Investopedia article (if there is one) or something on 130/30?

File: examples/README.md [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+Some stocks cannot be shorted.  In this case, the algorithm uses the leftover equity from the stocks that could not be shorted and shorts the stocks have already been shorted.
+
+The algorithm uses percent change in stock price over the past 10 minutes to rank the stocks, where the stocks the rose the most are longed and the ones that sunk the most are shorted.
```
Comment (ttt733) [inline examples/README.md]: "the stocks that rose"

File: examples/long-short/long-short.go [hunk 6]
```diff
@@ -0,0 +1,387 @@
...
+	}
+
+	// Wait for market to open.
+	fmt.Println("Waiting for market to open...")
+	for {
+		isOpen := alpacaClient.awaitMarketOpen()
+		if isOpen {
+			break
+		}
+		time.Sleep(2000 * time.Millisecond)
```
Comment (ttt733) [inline examples/long-short/long-short.go]: Any reason for this to be 2000 rather than 1000?

File: examples/README.md [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+
+## Mean Reversion
+
+This trading algorithm bases its strategy on a mean reversion theory, which essentially in the long run, the stock price will correct to the mean.  This means we'd want to execute trades when the stock price is below the running average, as the theory states that the stock price will eventually rise to the mean.  The algorithm does this by taking the 20 minute running average stock price of a given stock (in this case "AAPL") and longs or sells based on the average.  After every minute, the algorithm will re-evaluate the mean and see if adjustments to the position need to be made.  For more information on this strategy, you can read [this link](https://medium.com/automation-generation/a-simple-mean-reversion-stock-trading-script-in-c-fdd3d147af95) detailing a mean reversion strategy in C#.
+
+## Long-Short Equity
```
Comment (ttt733) [inline examples/README.md]: Maybe add something referencing the fact that this is essentially the opposite of the mean reversion strategy - it's betting on momentum instead.

**Comment by ttt733** [comment]
Haven't tested it but the code looks good to me, just the minor changes I mentioned to the readme.

File: examples/mean-reversion/mean-reversion.go [hunk 8]
```diff
@@ -0,0 +1,246 @@
+package main
```
Comment (ttt733) [inline examples/mean-reversion/mean-reversion.go]: Should we remove this file from this branch?

File: examples/long-short/long-short.go (hunk 6 above)
Comment (andrewkim316) [reply examples/long-short/long-short.go]: Should actually be 60000 to correspond to a minute, forgot to change it

File: examples/README.md (hunk 4 above)
Comment (andrewkim316) [reply examples/README.md]: Found one and linked it

**Pull request merged**